src/include/btree_inline.h

```diff
@@ -1485,21 +1485,16 @@ __wt_page_del_active(WT_SESSION_IMPL *session, WT_REF *ref, bool visible_all)
 }
 
 /*
- * __wt_btree_can_evict_dirty --
- *     Check whether eviction of dirty pages or splits are permitted in the current tree. We cannot
- *     evict dirty pages or split while a checkpoint is in progress, unless the checkpoint thread is
- *     doing the work. Also, during connection close, if we take a checkpoint as of a timestamp,
- *     eviction should not write dirty pages to avoid updates newer than the checkpoint timestamp
- *     leaking to disk.
+ * __wt_btree_in_checkpoint --
+ * Returns true if the session is performing a sync on its current tree.
```

Comment:
I think of this as a test for "Is the tree being flushed by another thread, because if that's the case I should leave it alone". I'm not sure what's the best way to say that.

Maybe phrase this as `Returns true if the current tree is not in checkpoint or the current thread is the checkpointing thread.`

But, that argues for a better function name, it's not just "in checkpoint". Ummmm, `__wt_btree_other_thread_checkpoint()`?

That's unwieldy, do you have a better idea?

Reply:
@keithbostic  What about `__wt_btree_syncing_by_other_session`?
I have also redesigned logic in the check, I think it's more readable this way, what do you think?

Reply:
This looks good to me --  thanks!